core/app/api/v0/c/[communitySlug]/site/[...ts-rest]/route.ts

```diff
@@ -226,6 +230,81 @@ const shouldReturnRepresentation = async () => {
 	return false;
 };
 
+/**
+ * manually parses the `?filters` query param.
+ * necessary because ts-rest only supports parsing object in query params
+ * if they're uri encoded.
+ *
+ * eg this does not fly
+ *  ```
+ * ?filters[community-slug:fieldName][$eq]=value
+ * ```
+ * but this does
+ *  ```
+ * ?filters=%7B%22%7B%22updatedAt%22%3A%20%7B%22%24gte%22%3A%20%222025-01-01%22%7D%2C%22field-slug%22%3A%20%7B%22%24eq%22%3A%20%22some-value%22%7D%7D`
+ * ```
+ *
+ * the latter is what a ts-rest client sends if `json-query: true`. we want to support both syntaxes.
```

Comment:
it's not so much wanting to support _both_ syntaxes, i just want to support the human readable one. turns out it's easier to support both.